src/components/OfferSwitch/OfferSwitch.tsx

```diff
@@ -0,0 +1,61 @@
+import classNames from 'classnames';
+import { useTranslation } from 'react-i18next';
+
+import styles from './OfferSwitch.module.scss';
+
+import type { Offer } from '#types/checkout';
+import Checkbox from '#components/Checkbox/Checkbox';
+import { formatLocalizedDate, formatPrice } from '#src/utils/formatting';
+import { useAccountStore } from '#src/stores/AccountStore';
+
+interface OfferSwitchProps {
+  isCurrentOffer: boolean;
+  pendingDowngradeOfferId: string;
+  offer: Offer;
+  selected: {
+    value: boolean;
+    set: React.Dispatch<React.SetStateAction<string | null>>;
+  };
+}
+
+const OfferSwitch = ({ isCurrentOffer, pendingDowngradeOfferId, offer, selected }: OfferSwitchProps) => {
+  const { t, i18n } = useTranslation('user');
+  const { customerPriceInclTax, customerCurrency, period } = offer;
+  const expiresAt = useAccountStore((state) => state.subscription?.expiresAt);
```

Comment:
Prevent using stores inside components. You can pass the `expiresAt` via props from the parent container. This keeps all store logic central and components "dumb".